protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -34,22 +71,51 @@ use crate::Error::MissingBytes;
 #[cfg(feature = "noise_sv2")]
 use crate::State;
 
-#[cfg(feature = "noise_sv2")]
-pub type StandardNoiseDecoder<T> = WithNoise<Buffer, T>;
+/// A decoded frame that could either be a regular or Noise-protected frame.
 pub type StandardEitherFrame<T> = EitherFrame<T, <Buffer as IsBuffer>::Slice>;
+
+/// A decoded Sv2 frame.
 pub type StandardSv2Frame<T> = Sv2Frame<T, <Buffer as IsBuffer>::Slice>;
+
+/// Standard decoder with Noise protocol support.
+#[cfg(feature = "noise_sv2")]
+pub type StandardNoiseDecoder<T> = WithNoise<Buffer, T>;
+
+/// Standard Sv2 decoder without Noise protocol support.
 pub type StandardDecoder<T> = WithoutNoise<Buffer, T>;
 
+/// Decoder for Sv2 frames with Noise protocol support.
```

Comment:
here we can add an example of how to create and use a codec. If is possible to do it vai a doc test would be fantastic.